answer/negativeAnswer/error.json

```diff
@@ -0,0 +1,28 @@
+{
+	"$schema": "http://json-schema.org/draft/2019-09/schema#",
+	"$id": "http://pop2020.dedis.ch/answer/negativeAnswer/error.json",
+	"title": "If an error occurred, the “result” field is absent and the error object must adhere to the following (extensible) format",
+	"description": "In case of negative answer, error generated by the client query",
+
+	"type": "object",
+	"properties": {
+		"code": { 
+			"description": "[Int] error code", 
+			"type": "integer", 
+			"minimum": -5, "maximum": -1,
+			"$comment": "Note: should be in { -5, -4, ..., -1 }",
+			"note": [
+				"0  operation was successful (should never be used)",
+				"-1 invalid action",
+				"-2 invalid resource (e.g. channel does not exist, channel was not subscribed to, etc.)",
+				"-3 resource already exists (e.g. lao already exists, channel already exists, etc.)",
+				"-4 request data is invalid (e.g. message is invalid), use the data field object to provide extra information",
+				"-5 access denied (e.g. subscribing to a “restricted” channel)"
+			]
+		},
+		"description": { "description": "[String] error description", "type": "string" }
+	},
+	"additionalProperties": false,
+	"required": ["code", "description"],
+	"$comment": "Note: this field is absent if the answer is positive"

```

Comment:
what does "this" refer to?

Reply:
it refers to the whole "error field" which is a field in a generic message (the entire file). I will change the comment to make it more clear!